packages/hydrooj/src/model/contest.ts

```diff
@@ -555,9 +567,10 @@ const ledo = buildContestRule({
         for (const pid of tdoc.pids) {
             row.push({
                 type: 'record',
-                value: tsddict[pid]?.penaltyScore?.toString() || '',
+                value: ((tsddict[pid]?.penaltyScore || 0) * ((tdoc.score?.[pid] || 100) / 100)).toString(),
```

Comment:
Maybe we need to keep the use of `score` / `orginalScore` uniform to represent both the contest score and the record score